lib/discover/file.js

```diff
@@ -0,0 +1,20 @@
+var fs = require('fs');
+
+module.exports.discover = function fileDiscover(urlObj, original, callback) {
```

Comment:
I generally don't like exporting functions directly or overwrite the hash of `module.exports`. Had a hard time debugging that once. But given that the chances of circular dependencies are low here we can change it to an exported function. Is there a specific reason to export the function directly?